app/services/merge_items_cart_service.rb

```diff
@@ -0,0 +1,44 @@
+class MergeItemsCartService
+  attr_reader :session_order, :current_user, :exist_user_order, :user_order
+
+  def initialize(session_order, current_user)
+    @session_order = session_order
+    @current_user = current_user
+    @exist_user_order = Order.where(user_id: current_user.id).order(:updated_at).last
+    @user_order = exist_user_order if exist_user_order && !exist_user_order.complete?
+  end
+
+  def call
+    return change_items_order_id_to_current if user_order
+
+    add_current_order_session_user_id
+  end
+
+  def add_current_order_session_user_id
```

Comment:
you should have helper method `current_order` in ApplicationController instead of this